kore/src/Kore/Internal/Predicate.hs

```diff
@@ -788,6 +788,8 @@ hasFreeVariable
     -> Bool
 hasFreeVariable variable = isFreeVariable variable . freeVariables
 
+-- TODO(Ana): change to Substitution.Assignment -> Predicate;
+-- this would require moving Assignment to a new module
```

Comment:
That's because `Kore.Internal.Substitution` imports `Kore.Internal.Predicate`? We would probably move this function instead; it's the implementation of `instance From (Assignment variable) (Predicate variable)`.